napari_brightness_contrast/_dock_widget.py

```diff
@@ -287,31 +283,28 @@ def value_changed():
         self.layout().addWidget(slider)
         self.layout().addWidget(lbl_max)
 
-def histogram(layer, num_bins : int = 256, minimum = None, maximum = None, use_cle=True):
+def histogram(layer, num_bins : int = 256, minimum = None, maximum = None):
     """
     This function determines a histogram for a layer and caches it within the metadata of the layer. If the same
     histogram is requested, it will be taken from the cache.
     :return:
     """
-    if "bc_histogram_num_bins" in layer.metadata.keys() and "bc_histogram"  in layer.metadata.keys():
+    if "bc_histogram_num_bins" in layer.metadata.keys() and "bc_histogram" in layer.metadata.keys():
         if num_bins == layer.metadata["bc_histogram_num_bins"]:
             return layer.metadata["bc_histogram"]
 
-    data = layer.data
-    if "dask" in str(type(data)): # ugh
-        data = np.asarray(data)
-    intensity_range = None
-    if minimum is not None and maximum is not None:
-        intensity_range = (minimum, maximum)
-
-    if use_cle:
-        try:
-            import pyclesperanto_prototype as cle
-            hist = np.asarray(cle.histogram(data, num_bins=num_bins, minimum_intensity=minimum, maximum_intensity=maximum, determine_min_max=False))
-        except ImportError:
-            use_cle = False
-    if not use_cle:
-        hist, _ = np.histogram(data, bins=num_bins, range=intensity_range)
+
+    data = np.asarray(layer._data_view)
```

Comment:
```suggestion
    data = layer.data
    if "dask" in str(type(data)): # ugh
        data = np.asarray(data)
```
See my other comment...

Reply:
I would still use the private layer._data_view for now, I'm affraid using layer.data will bring back the slowness and incompatibilities with big images (not tested yet)